osgar/test_logger.py

```diff
@@ -230,5 +241,52 @@ def test_large_blocks_with_time_overflow(self):
             self.assertEqual(dt, timedelta(hours=2))
         os.remove(filename)
 
+    def test_no_eof(self):
+        # similar to tail -f (--follow)
+        with LogWriter(prefix='tmpB', note='test_EOF') as log:
+            filename = log.filename
+            t1 = log.write(1, b'\x01'*100)
+            time.sleep(0.001)
+            t2 = log.write(1, b'\x02'*100)
+            time.sleep(0.001)
+            t3 = log.write(1, b'\x03'*100000)
+
+        with LogReader(filename) as log:
+            dt, channel, data = next(log.read_gen(only_stream_id=1))
+            self.assertEqual(data, b'\x01'*100)
+            dt, channel, data = next(log.read_gen(only_stream_id=1))
+            self.assertEqual(data, b'\x02'*100)
+
+        partial = filename + '.part'
+        with open(filename, 'rb') as f_in:
+            with open(partial, 'wb') as f_out:
+                f_out.write(f_in.read(100))
+        
+        with LogReader(partial) as log:
+            with self.assertRaises(AssertionError):
+                dt, channel, data = next(log.read_gen(only_stream_id=1))
+
+        proc = Timer(0.1, delayed_copy, [filename, partial, 100])
+        proc.start()
+        with LogReader(partial, follow=True) as log:
+            dt, channel, data = next(log.read_gen(only_stream_id=1))
+            self.assertEqual(data, b'\x01'*100)
+        proc.join()
+
+        partial = filename + '.part'
+        with open(filename, 'rb') as f_in:
+            with open(partial, 'wb') as f_out:
+                f_out.write(f_in.read(100000))
+
+        with LogReader(partial) as log:
+            dt, channel, data = next(log.read_gen(only_stream_id=1))
+            self.assertEqual(data, b'\x01'*100)
+            dt, channel, data = next(log.read_gen(only_stream_id=1))
+            self.assertEqual(data, b'\x02'*100)
+            with self.assertRaises(AssertionError):
+                dt, channel, data = next(log.read_gen(only_stream_id=1))
+
+        os.remove(partial)
+        os.remove(filename)
```

Comment:
Just a side note only partially related to this PR: I think the tests should be using a temporary directory.

Reply:
OK - I used to analyze failing tests from available intermediate files, but it is no longer needed. Ack.

Reply:
Autocreating a temporary directory for the test will be left to a different time, right?

Reply:
Yes please